packages/workflows/src/handlers/product/update-products-prepare-data.ts

```diff
@@ -0,0 +1,80 @@
+import { ProductDTO, SalesChannelDTO, WorkflowTypes } from "@medusajs/types"
+
+import { WorkflowArguments } from "../../helper"
+
+type ProductWithSalesChannelsDTO = ProductDTO & {
+  sales_channels?: SalesChannelDTO[]
+}
+
+export type UpdateProductsPreparedData = {
+  originalProducts: ProductWithSalesChannelsDTO[]
+  productHandleAddedChannelsMap: Map<string, string[]>
+  productHandleRemovedChannelsMap: Map<string, string[]>
+}
+
+export async function updateProductsPrepareData({
+  container,
+  context,
+  data,
+}: WorkflowArguments<WorkflowTypes.ProductWorkflow.UpdateProductsWorkflowInputDTO>): Promise<UpdateProductsPreparedData> {
+  const ids = data.products.map((product) => product.id)
+
+  const productHandleAddedChannelsMap = new Map<string, string[]>()
+  const productHandleRemovedChannelsMap = new Map<string, string[]>()
+
+  const productService = container.resolve("productService")
+  const productServiceTx = productService.withTransaction(context.manager)
+
+  const products = await productServiceTx.list(
```

Comment:
Add a TODO comment here to use the RemoteQuery.
Sales Channel will have to be added on the medusa/joiner-configs to be joined